pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -1624,9 +1620,35 @@ async def run(self):
                                 d_content.content.files = result.files
                             if incorrect_changelogs or incorrect_modular_relation:
                                 log.debug("Updated data for package {}".format(result.nevra))
-                                await sync_to_async(result.save)()
+                                saved_rslts_pids.append(result.pulp_id)
+                                saved_rslts_by_pid[result.pulp_id] = result
                         # ==================================================================
                         d_content.content = result
 
+            # Save results in guaranteed-pid-order
+            saved_rslts_pids.sort()
+            for pid in saved_rslts_pids:
+                await sync_to_async(saved_rslts_by_pid[pid].save)()
+
+            # touch all affected content, **excluding** anything we just saved
```

Comment:
But the idea here is that we're touching things we see during the sync, and the packages we are re-saving still fall into that category too

Reply:
We are explicitly **excluding** from the touch() anything we've modified/explicitly saved above. Explicit-save updates timestamp_of_interest along the way (which is part of why we're deadlocking in this particular pathological case).

Reply:
Ah ok.